Benchmarks.hs

```diff
@@ -141,6 +142,16 @@ sourceLG v = msum $ map return [v..v+value]
 runIOLG :: LG.LogicT IO Int -> (Int -> LG.LogicT IO Int) -> IO ()
 runIOLG s t = LG.observeAllT (s >>= t) >> return ()
 
+-------------------------------------------------------------------------------
+-- vector
+-------------------------------------------------------------------------------
+
+sourceV :: MonadIO m => Int -> V.Stream m Int
+sourceV v = V.fromList [v..v+value]
```

Comment:
Any particular reason why you are passing `v` as an argument (like `sourceM`) rather than obtaining it right here using `getRandom` like in the case of `sourceA` or `sourceS` etc.

Reply:
I couldn't find a good way to go from `IO (Stream IO Int)` to `Stream IO Int`.

Reply:
Yeah, it seems `Stream m a` is not a Monad. Perhaps we do not need a `MonadIO` constraint here, just `Monad` should be enough since we are not using `getRandom`. I can fix that if needed. I will also add a `-Wredundant-constraints` flag to GHC if it can find out some such stuff.